source/common/buffer/buffer_impl.cc

```diff
@@ -43,71 +56,213 @@ void OwnedImpl::add(const Instance& data) {
 }
 
 void OwnedImpl::prepend(absl::string_view data) {
-  evbuffer_prepend(buffer_.get(), data.data(), data.size());
+  uint64_t size = data.size();
```

Comment:
More drive-by; it would be super-awesome to have a buffer fuzzer. You could use https://github.com/envoyproxy/envoy/blob/master/test/common/http/header_map_impl_fuzz_test.cc as an example of how to write this kind of data structure fuzzer. The interesting thing here is you could easily state correctness conditions I think, so you might be able to do a stronger than "it doesn't crash" fuzzer. Fine if you want to create a TODO or file an issue, but I think this would increase confidence in implementation correctness.